File: src/logalytics/model/schema/log_schema.py
from typing import Union, List, Dict
from typing.re import Pattern
import re

from logalytics.model.types import GroupName, EntryItemParser


class LogSchema:
    def __init__(self,
                 pattern: Union[GroupName, Pattern],
                 groups: List[GroupName],
                 parsers: Dict[GroupName, EntryItemParser]):
        self._pattern = re.compile(pattern)
        self._groups = groups
        self._parsers = parsers

    @property
    def pattern(self) -> Union[GroupName, Pattern]:
        return self._pattern

    @property
    def groups(self) -> List[GroupName]:
        return self._groups

    @property
    def parsers(self) -> Dict[GroupName, EntryItemParser]:
        return self._parsers

File: src/main/java/logalytics/config/LogSchema.java
package logalytics.config;

import java.util.List;
import java.util.regex.Pattern;

public class LogSchema {
    private final String fileName;
    private final String regex;
    private final List<String> groups;

    public LogSchema(String fileName, String regex, List<String> groups) {
        this.fileName = fileName;
        this.regex = regex;
        this.groups = groups;
    }

    public String fileName() {
        return fileName;
    }

    public Pattern regex() {
        return Pattern.compile(this.regex);
    }

    public List<String> groups() {
        return this.groups;
    }
}
